### PR TITLE
fixes barnyard only being able to curse humans

### LINKED ISSUE #41721: you can only barnyard curse humans, not any other species
it wants it to match type, but only lists /mob/living/carbon/human and monkeys

### PR DESCRIPTION
**Pull request by tralezab**
fixes #41721

i might need to test this first

### CODE REVIEW

File: code/modules/spells/spell_types/barnyard.dm [hunk 1]
```diff
@@ -23,7 +23,11 @@
 
 	var/mob/living/carbon/target = targets[1]
 
-	if(!(target.type in compatible_mobs))
```
Comment (tralezab) [inline code/modules/spells/spell_types/barnyard.dm]: it checked the specific type but not sub types which messed it up, or something like that

**Comment by adamkad1** [comment]
will it kill plasmamems?

**Comment by davethwave** [comment]
It would if the mask doesn't allow internals

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Use the typecache function family, change `compatible_mobs` to a typecache

**Comment by tralezab** [comment]
@Cyberboss this should work but you should double check just in case

File: code/modules/spells/spell_types/barnyard.dm [hunk 2]
```diff
@@ -12,7 +12,7 @@
 	range = 7
 	cooldown_min = 30
 	selection_type = "range"
-	var/list/compatible_mobs = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
+	var/list/compatible_mobs_typecache = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
```
Comment (SpaceManiac) [inline code/modules/spells/spell_types/barnyard.dm]: I believe you will need to do this, in order for the `is_type_in_typecache` check to work correctly:
```suggestion
	var/list/compatible_mobs_typecache = typecacheof(/mob/living/carbon/human, /mob/living/carbon/monkey)
```
I suggest testing though.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Comment by tralezab** [comment]
i think you were right, let me try this

**Comment by tralezab** [comment]
image
confirmed working now

**Comment by ExcessiveUseOfCobblestone** [comment]
This will kill plasmamen, as none of the barn masks allow for internal usage.

**Comment by SpaceManiac** [comment]
It did before, it was just inconsistent (ex-humans transformed into plasmamen were vulnerable, for instance).

**Comment by ExcessiveUseOfCobblestone** [comment]
rip

**Comment by tralezab** [comment]
sucks to be a plasmaman, people don't pick barnyard often anyways.

**Comment by pizzatiger** [comment]
I see no problem with it killing plasma-men seeing how plasma men are immune to hive-minds heart-attack power

**Pull request merged**